packages/server/src/fhircast/routes.ts

```diff
@@ -99,9 +101,39 @@ protectedCommonRoutes.post(
     }
 
     const topic = req.body['hub.topic'];
-    const subscriptionEndpoint = topic; // TODO: Create separate subscription endpoint for topic
-    const config = getConfig();
+    let subscriptionEndpoint: string;
+    try {
+      const results = await getRedis()
+        // Multi allows for multiple commands to be executed in a transaction
+        .multi()
+        // Sets the endpoint key for this topic if it doesn't exist
+        .setnx(`medplum:fhircast:topic:${topic}:endpoint`, generateId())
+        // Gets the endpoint, either previously generated endpoint secret or the newly generated key if a previous one did not exist
+        .get(`medplum:fhircast:topic:${topic}:endpoint`)
+        // Executes the transaction
+        .exec();
+
+      if (!results) {
+        throw new Error('Redis returned no results while retrieving endpoint for this topic');
+      }
+
+      assert(results.length === 2, 'Redis did not return 2 command results for FHIRcast endpoint retrieval');
 
+      const [err, result] = results[1];
+      if (err) {
+        throw err;
+      }
+      subscriptionEndpoint = result as string;
+    } catch (err) {
+      sendOutcome(res, serverError(new Error('Failed to get endpoint for topic')));
```

Comment:
Using `serverError` will return HTTP 500.  Maybe `badRequest` for HTTP 400 would be better?

Reply:
In this case it would actually be good if it returned a 500 since the only time this case should be hit is if something happens with Redis

Reply:
>  it would actually be good if it returned a 500 since the only time this case should be hit is if something catastrophic happens to Redis

Hm, can you check my logic here?  My interpretation is that a well crafted malicious HTTP request with an unknown topic ID would trigger a 500, which would wake people up in the middle of the night.  If that's correct, I think we should at least handle that one case and throw a 404.

Reply:
This route is like an upsert kinda thing,
That's why we call `setnx` before `get` in this transaction. 

The logic is: 
1) We conditionally set the topic's url to be a new random UUID if it doesn't exist. If it does, skip this step.
2) We get the value of the topic's url

If this fails for some reason beyond what is normal Redis operation (thinking timeouts or something like that, not actually sure what cases would trigger this but they are outside of healthy)